src/libpmemobj/tx.c

```diff
@@ -1069,11 +1069,9 @@ pmemobj_tx_add_snapshot(struct tx *tx, struct tx_range_def *snapshot)
 		if (action == NULL)
 			return -1;
 
-		/* first entry of the first ulog */
-		struct ulog_entry_base *e =
-			(struct ulog_entry_base *)tx->lane->layout->undo.data;
+		uint64_t *n = &tx->lane->layout->undo.gen_num;
```

Comment:
update the comment above if. We no longer clear the first entry but rather bump the `gen_num`.